src/training_core.cpp

```diff
@@ -10,16 +10,13 @@ using std::endl;
 
 const bool use_multiset_logsum = 
 #ifndef USE_MULTISET_LOGSUM
```

Comment:
It would be better to `#define USE_MULTISET_LOGSUM` somewhere above here rather than rewrite this ifdef. In the new version if `USE_MULTISET_LOGSUM` is defined it won't compile.

Reply:
would it be better to just use `const bool use_multiset_logsum = true;`

Reply:
I can't see anywhere else that USE_MULTISET_LOGSUM is used, so that would be fine.